docs/rotate-git-crypt.md

```diff
@@ -0,0 +1,55 @@
+# Rotating the git-crypt key
+
+We use `git-crypt` for securing secrets committed to `git`. It uses a symmetric encryption key which is then encrypted for users using their GPG keys.
+
+When rotating the `git-crypt` symmetric key, you should follow the steps below:
+
+1. Ensure you have unlocked the repository: `git-crypt unlock`
+
+2. Make a list of the current users:
+  ```
+  ls .git-crypt/keys/default/0/*.gpg | sed -E 's/^.+\/([A-Z0-9]{40}).gpg$/\1/' > git-crypt-users
+  ```
+
+2. Make a list of the encrypted files:
+  ```
+  git-crypt status -e | sed -E 's/^.+encrypted:[[:space:]]+([^[:space:]]+)$/\1/' > git-crypt-files
+  ```
+
+3. Remove `git-crypt` configuration:
+  ```
+  rm -rf .git-crypt .git/git-crypt
+  ```
+
+4. Re-initialise `git-crypt`,
+  ```
+  git-crypt init
+  ```
+  make sure you have all the users' keys,
+  ```
+  cat git-crypt-users | xargs -I{} gpg --recv-key {}
+  ```
+  and add all the users back,
+  ```
+  cat git-crypt-users | xargs -I{} git-crypt add-gpg-user -n --trusted {}
+  ```
```

Comment:
👍 Bonus points for this use of xargs! One of my favourite and most overlooked unix tools.